module/VuFind/src/VuFind/Db/Row/Search.php

```diff
@@ -160,4 +171,236 @@ public function getUnsubscribeSecret(HMAC $hmac, $user)
         ];
         return $hmac->generate(array_keys($data), $data);
     }
+
+    /**
+     * Get identifier (returns null for an uninitialized or non-persisted object).
+     *
+     * @return ?int
+     */
+    public function getId(): ?int
+    {
+        return $this->id ?? null;
+    }
+
+    /**
+     * Get user.
+     *
+     * @return int
+     */
+    public function getUserId(): int
+    {
+        return $this->user_id
+        ? $this->getDbServiceManager()->get(UserServiceInterface::class)->getUserById($this->user_id)
```

Comment:
Since user_id is not allowed to be null in the table, you don't need this ternary check. (I realize that something weird may happen if somebody calls the getter on an uninitialized object, but they shouldn't do that -- we could add throwing an exception if you want to explicitly address that possibility).